### DISCUSSION TITLE
High CPU usage when a DAG is modified

### DISCUSSION THREAD

**Discussion by FelipeArisi** [apache/airflow#26421] [General]
### Apache Airflow version

Other Airflow 2 version

### What happened

Whenever I update the DAGs, the scheduler's CPU usage increases by exactly 3 hours.
Here is an example using grafana.

![image](https://user-images.githubusercontent.com/15203706/190402811-53c9d007-82bb-45c8-ad2c-91c288f7b202.png)

I'm using a linux server and airflow is running with docker-compose.
Image: apache/airflow:2.2.1-python3.9

To change the DAG I'm using the command 'docker cp'

Config:
    AIRFLOW__SCHEDULER__DAG_DIR_LIST_INTERVAL: 600
    AIRFLOW__SCHEDULER__MIN_FILE_PROCESS_INTERVAL: 600
    AIRFLOW__SCHEDULER__SCHEDULER_IDLE_SLEEP_TIME: 5
    AIRFLOW__CORE__MIN_SERIALIZED_DAG_UPDATE_INTERVAL: 600
    AIRFLOW__CORE__MIN_SERIALIZED_DAG_FETCH_INTERVAL: 300
    AIRFLOW__WEBSERVER__WORKERS: 2 
    AIRFLOW__WEBSERVER__WORKER_REFRESH_INTERVAL: 1800
    AIRFLOW__WEBSERVER__WEB_SERVER_WORKER_TIMEOUT: 300 

### Operating System

linux/amd64

### Deployment

Docker-Compose

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template!

**Comment by potiuk** [comment]
This is at most a discussion not an issue, especially that you have not specify neither expected behaviour nor ways to reproduce. And yo have not stated what problem you are trying to solve. 

Please avoid it in the future.

It is quite normal to have CPU increase that when something happens (parsing and syncing DAGs) more CPU is used.

Why do you think it is wrong? What made you think it is?
What  processes are taking the CPU? Did you make any investigation as to that? What other resources did you observe? what your logs are saying?  How many DAgs you have and how long does it take scheduier /Dag file Processor to parse them (you can find all this information in the logs).

For all this questions you are the only person who can answer them. If you want to ask some questions or when you think that Airflow should behave differently (and how, with some reasonable justifications please).
